interview-questions.md

```diff
@@ -8,41 +8,39 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The map method uses the original content of the array and produces the same array just after it has undergone a function/action to it. While the filter method is similiar because it takes the same array but with its functionality edits out certain elements within an array based on the parameters set. 
 
+  Researched answer: Both the map and the filter methods focus primarily the with array and will produce an output that is essentially an updated version of the original array. The difference is with the map method will be altering each element within the array but still return the same length of that array. Conversely, with the filter method inspect each element of an array and based on the parameters given it will eliminate certain ones in the output and thus create an array that is shorter in length than the original. 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: A function is lines of code written in a text editor to perform certain tasks. A method is a type of function that deals exclusively with objects. Function and method have an interwined relationship because a method will always be a function but a function doesn't necessarily have to be a function; similiar to a square is a type of rectangle but not all rectangles are squares. 
 
+  Researched answer: A function is a sequence of written code that are designed to perform specific sets of tasks. It is defined by a function keyword followed and optional parameters (depending on the circumstance) and is encapsulated by Curly brakets. Functions can be associated with objects or not, however when an object is involved this type of function is known as a method. With a method contains that can be identified by the name of its object and and its method name (specific name given that often defines the task action) through the use of dot notation or square braket notation. Lastly, just as classical is a type of music but not all music of course, method is simply one type of function.
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is breaking down an object into specific output. So if you had a function with that contained alot of personal informatin like names, address, emails, etc and need to limit all that to just one quick output of a particular name and/or address, you would log that in and the return should produce just that information. 
 
-  Researched answer:
+  Researched answer: Object destructuring is a unique way of redefining the available variables; meaning it allows the programmer to grab objects or arrays and break down the connected values into individualized variables. This is useful because when comes to nested properties, you end up saving a lot of time by using object destructuring due to being able to efficiently alter and pull object properties. 
```

Comment:
Another week, another set of great interview question responses, Steven!  These are fantastic.

Reply:
Much appreciated Tricia.